### PR TITLE
Linen activations import

### PR DESCRIPTION
**Pull request by onnoeberhard**
# What does this PR do?

Addresses #2087 

These changes make `flax.linen` more consistent with `jax.nn` in the sense that all functions one might import from `jax.nn` can now directly be imported from `flax.linen` as well. This PR includes 3 new imports from `jax.nn` into `flax.linen.activations` and additional imports from `flax.linen.activations` into `flax.linen.__init__`.

The only import left out is `jax.nn.standardize` which will replace `jax.nn.normalize` (deprecated) in the future, but had not been introduced in JAX version 0.3 (Flax's dependency).

Tagging @marcvanzee

## Checklist
- [x] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).
- [x] This change is discussed in a Github discussion: #2087 
- [ ] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).
- [ ] This change includes necessary high-coverage tests.
      (No quality testing = no merge!)

### CODE REVIEW

**Review by marcvanzee — APPROVED** [review]
Thanks a lot!

**Comment by marcvanzee** [comment]
Note: I squashed your commits into 1 to keep our commit tree clean.

File: flax/linen/__init__.py [hunk 1]
```diff
@@ -23,10 +23,19 @@
...
   glu as glu,
+  hard_sigmoid as hard_sigmoid,
+  hard_silu as hard_silu,
+  hard_swish as hard_swish,
+  hard_tanh as hard_tanh,
   leaky_relu as leaky_relu,
   log_sigmoid as log_sigmoid,
   log_softmax as log_softmax,
+  logsumexp as logsumexp,
+  normalize as normalize,
```
Comment (levskaya) [inline flax/linen/__init__.py:34]: could we add the `standardize` alias for `normalize` as well? (this name change follows stats literature naming)

Comment (onnoeberhard) [reply flax/linen/__init__.py:34]: I initially added it, but then removed it again when I saw that it had only been added to JAX very recently (not in version 0.3, which I understand to be Flax's dependency).

I guess we could do something like `try: import standardize; except: pass`, but this might be bad style. Or we could define the alias ourselves.

Comment (onnoeberhard) [reply flax/linen/__init__.py:34]: What do you think would be the best option here?

Comment (marcvanzee) [reply flax/linen/__init__.py:34]: It seems cleaner to create an issue to add this once JAX rolls out a new version.

**Comment by marcvanzee** [comment]
Sorry, I forgot about this PR. I will get it in ASAP and create an issue for adding the additional activation functions once JAX rolls out a new version.

File: flax/linen/__init__.py (hunk 1 above)
Comment (marcvanzee) [reply flax/linen/__init__.py:34]: Added #2119
